vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -282,26 +282,36 @@ fire_and_forget WinRTHttpResource::PerformSendRequest(HttpRequestMessage &&reque
         reader.UnicodeEncoding(UnicodeEncoding::Utf8);
       }
 
-      // #9510 - 10mb limit on fetch
-      co_await reader.LoadAsync(10 * 1024 * 1024);
-      auto length = reader.UnconsumedBufferLength();
+      uint64_t segmentSize = 10 * 1024 * 1024;
+      auto contentLengthHeader = response.Content().Headers().ContentLength();
 
-      if (coReqArgs->IsText) {
-        std::vector<uint8_t> data(length);
-        reader.ReadBytes(data);
-        string responseData = string(Common::Utilities::CheckedReinterpretCast<char *>(data.data()), data.size());
+      uint64_t contentSize = segmentSize;
+      if (contentLengthHeader) {
+        contentSize = contentLengthHeader.Value();
+      }
 
-        if (self->m_onData) {
-          self->m_onData(coReqArgs->RequestId, std::move(responseData));
-        }
-      } else {
-        auto buffer = reader.ReadBuffer(length);
-        auto data = CryptographicBuffer::EncodeToBase64String(buffer);
-        auto responseData = to_string(std::wstring_view(data));
+      // #9510 - We currently accumulate all incoming request data in 10MB chunks.
+      string responseData;
+      auto length = segmentSize;
+      do {
+        co_await reader.LoadAsync(static_cast<uint32_t>(segmentSize));
+        length = reader.UnconsumedBufferLength();
 
-        if (self->m_onData) {
-          self->m_onData(coReqArgs->RequestId, std::move(responseData));
+        if (coReqArgs->IsText) {
+          auto data = std::vector<uint8_t>(length);
+          reader.ReadBytes(data);
+
+          responseData += string(Common::Utilities::CheckedReinterpretCast<char *>(data.data()), data.size());
+        } else {
+          auto buffer = reader.ReadBuffer(static_cast<uint32_t>(length));
+          auto data = CryptographicBuffer::EncodeToBase64String(buffer);
+
+          responseData += to_string(std::wstring_view(data));
         }
+      } while (length > 0);
```

Comment:
I don't see contentSize being read.... It looks like this do/while loop will read only up to segmentSize (since it's based on length)?

Reply:
Ah, good catch.
That's stale code.

The WinRT APIs allow to keep reading a given bidder size until the read length is zero.

We don't need to read the content length to tell when we're done.

I'll delete before merging.